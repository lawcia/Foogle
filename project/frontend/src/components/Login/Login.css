.Login {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 10vh;
}

.Login__form {
  width: 300px;
  display: flex;
  flex-direction: column;
  padding: 16px;
  margin: 16px 0;
}

.Login__input {
  margin-top: 4px;
  margin-bottom: 16px;
}

.Login__logo {
  width: 200px;
}

.Login__heading {
  font-size: 30px;
  text-align: center;
}

.Login__label {
  font-weight: 600;
}

.Login--error {
  background-color: #ffcdd2;
  color: #630700;
  padding: 4px;
  width: 268px;
  text-align: center;
  margin-top: 16px;
  border-radius: 4px;
  border: 1px solid #630700;
}