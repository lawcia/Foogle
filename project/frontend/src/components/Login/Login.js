import React, { useState } from "react";
import Button from "../Buttons/Button";
import logo from "../../images/foogle.png";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { loginUser } from "../../redux/actions/authActions";
import "./Login.css";

export const Login = ({ error, loginUser }) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();
    loginUser({ username, password }).catch((error) => console.log(error));
  };

  return (
    <div className="Login">
      <img className="Login__logo" src={logo} alt="foogle logo" />
      <h3 className="Login__heading">Welcome back</h3>
      {error && error.length > 0 && (
        <div className="Login--error">
          {error.map((error) => (
            <p>{error}</p>
          ))}
        </div>
      )}
      <form className="Login__form" onSubmit={handleSubmit}>
        <label className="Login__label" htmlFor="username">
          Username
        </label>
        <input
          className="input input-m Login__input"
          onChange={(e) => setUsername(e.target.value)}
          type="text"
          name="username"
          value={username}
        />
        <label className="Login__label" htmlFor="password">
          Password
        </label>
        <input
          className="input input-m Login__input"
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          name="password"
          value={password}
        />
        <Button value="Log in" colour="blue" />
      </form>
      <div>
        <p>
          New to foogle? <Link to="/signup">Create an account</Link>
        </p>
      </div>
    </div>
  );
};

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  loginError: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  return {
    error: state.authReducer.loginError,
  };
};

const mapDispatchToProps = {
  loginUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
