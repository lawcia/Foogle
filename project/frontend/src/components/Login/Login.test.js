import React from "react";
import { Login } from "./Login";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

describe("<Login />", () => {
  it("should have div", () => {
    const wrapper = shallow(<Login error="" />);
    expect(wrapper.find("div.Login").length).toEqual(1);
  });

  it("should match default snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Login error=""/>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match error snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Login error="Something went wrong" />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
