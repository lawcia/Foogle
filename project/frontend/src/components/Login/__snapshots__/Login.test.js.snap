// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match default snapshot 1`] = `
<div
  className="Login"
>
  <img
    alt="foogle logo"
    className="Login__logo"
    src="test-file-stub"
  />
  <h3
    className="Login__heading"
  >
    Welcome back
  </h3>
  <form
    className="Login__form"
    onSubmit={[Function]}
  >
    <label
      className="Login__label"
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="input input-m Login__input"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="Login__label"
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="input input-m Login__input"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="button is-primary blue"
    >
      Log in
    </button>
  </form>
  <div>
    <p>
      New to foogle? 
      <a
        href="/signup"
        onClick={[Function]}
      >
        Create an account
      </a>
    </p>
  </div>
</div>
`;

exports[`<Login /> should match error snapshot 1`] = `
<div
  className="Login"
>
  <img
    alt="foogle logo"
    className="Login__logo"
    src="test-file-stub"
  />
  <h3
    className="Login__heading"
  >
    Welcome back
  </h3>
  <div
    className="Login--error"
  >
    <p>
      Something went wrong
    </p>
  </div>
  <form
    className="Login__form"
    onSubmit={[Function]}
  >
    <label
      className="Login__label"
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="input input-m Login__input"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="Login__label"
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="input input-m Login__input"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="button is-primary blue"
    >
      Log in
    </button>
  </form>
  <div>
    <p>
      New to foogle? 
      <a
        href="/signup"
        onClick={[Function]}
      >
        Create an account
      </a>
    </p>
  </div>
</div>
`;
